ResourceLeak.java

```diff
@@ -0,0 +1,66 @@
+package com.mariten.kanatools;
+
+import com.sun.org.slf4j.internal.LoggerFactory;
+
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+public class ResourceLeak {
+	private boolean active;
+    public void setActive(boolean b)
+          {
+            this.active = b;
+          }
+
+    LoggerFactory logger = new LoggerFactory();
+	String count;
+    public void process() {
+        SecureRandom secureRandom = new SecureRandom();
+	     System.runFinalizersOnExit(true);
+        try {
+            PrintWriter out = new PrintWriter("");
+            out.println("the text");
+System.out.println("");
+System.out.println("");   
+            out.close();
+            Cipher c1 = Cipher.getInstance("DES");
+        } catch (IOException | NoSuchAlgorithmException | NoSuchPaddingException e) {
+            System.err.println();
+System.err.println();
+
+       
+
+        }
+
+    }
+    public void process2() {
+        try {
+            PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("out.txt", true)));
+            out.println("the text");
+        } catch (IOException e) {
+            logger.error("Resource is not closed anywhere.",e);
+        }
+    }
+
+
+	  public void demo(String name, String surName,String count,String preeti) {
+		String surName = surName;
```

Comment:
I detect that this code is problematic. According to the [Dodgy code (STYLE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dodgy-code-style), [SA: Self assignment of local variable (SA_LOCAL_SELF_ASSIGNMENT)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-assignment-of-local-variable-sa-local-self-assignment).
This method contains a self assignment of a local variable; e.g.
Such assignments are useless, and may indicate a logic error or typo.